### PR TITLE
DTSPB-3405 CW Escalation state

### PR DESCRIPTION
**Pull request by MarcusHMCTS**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DTSPB-3405

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by IswaryaPepakayala — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/probate/model/ccd/raw/response/ResponseCaseDataParent.java [hunk 1]
```diff
@@ -75,6 +75,9 @@ public class ResponseCaseDataParent {
     protected final String solsReviewSOTConfirmCheckbox2Names;
     protected final String taskList;
     protected final String escalatedDate;
+    protected final String cwEscalatedDate;
+    protected final String caseWorkerEscalationDate;
```
Comment (FeliTam) [inline src/main/java/uk/gov/hmcts/probate/model/ccd/raw/response/ResponseCaseDataParent.java]: Two fields for escalated date
Prob one of them should be resolveCaseWorkerEscalationState?

Comment (MarcusHMCTS) [reply src/main/java/uk/gov/hmcts/probate/model/ccd/raw/response/ResponseCaseDataParent.java]: oh...thanks cwEscalatedDate deleted

File: src/main/java/uk/gov/hmcts/probate/transformer/CallbackResponseTransformer.java [hunk 2]
```diff
@@ -438,6 +438,14 @@ public CallbackResponse resolveStop(CallbackRequest callbackRequest) {
         return transformResponse(responseCaseDataBuilder.build());
     }
 
+
+    public CallbackResponse resolveCaseWorkerEscalationState(CallbackRequest callbackRequest) {
```
Comment (DelythJustice) [inline src/main/java/uk/gov/hmcts/probate/transformer/CallbackResponseTransformer.java:442]: maybe rename to set so it's clear this is a set not a resolve only

File: src/main/java/uk/gov/hmcts/probate/service/CaseEscalatedService.java [hunk 3]
```diff
@@ -15,4 +15,12 @@ public class CaseEscalatedService {
     public void caseEscalated(CaseDetails caseDetails) {
         caseDetails.getData().setEscalatedDate(LocalDate.now());
     }
+
+    public void caseWorkerEscalated(CaseDetails caseDetails) {
+        caseDetails.getData().setCaseWorkerEscalationDate(LocalDate.now());
+    }
+
+    public void resolveCaseWorkerEscalated(CaseDetails caseDetails) {
```
Comment (DelythJustice) [inline src/main/java/uk/gov/hmcts/probate/service/CaseEscalatedService.java]: don't you want to set the state here too?

**Review by DelythJustice — CHANGES_REQUESTED** [review]

File: ccdImports/configFiles/CCD_Probate_Backoffice/CaseField.json [hunk 4]
```diff
@@ -426,6 +426,9 @@
...
   {"LiveFrom": "01/01/2017", "CaseTypeID": "GrantOfRepresentation", "ID": "grantStoppedDate", "Label": "Grant stopped date", "FieldType": "Date", "SecurityClassification": "Public"},
   {"LiveFrom": "01/01/2017", "CaseTypeID": "GrantOfRepresentation", "ID": "escalatedDate", "Label": "Case escalated date", "FieldType": "Date", "SecurityClassification": "Public"},
+  {"LiveFrom": "01/01/2017", "CaseTypeID": "GrantOfRepresentation", "ID": "caseWorkerEscalationDate", "Label": "Case worker escalation date", "FieldType": "Date", "SecurityClassification": "Public"},
+  {"LiveFrom": "01/01/2017", "CaseTypeID": "GrantOfRepresentation", "ID": "resolveCaseWorkerEscalationDate", "Label": "Resolve Case worker escalation date", "FieldType": "Date", "SecurityClassification": "Public"},
+  {"LiveFrom": "01/01/2017", "CaseTypeID": "GrantOfRepresentation", "ID": "resolveCaseWorkerEscalationState", "Label": "Resolve Case worker escalation state to?", "FieldType": "FixedList", "FieldTypeParameter": "resolveEscalationStateFixedList", "SecurityClassification": "Public"},
```
Comment (DelythJustice) [inline ccdImports/configFiles/CCD_Probate_Backoffice/CaseField.json:431]: Is this label supposed to have the ?

File: src/main/java/uk/gov/hmcts/probate/transformer/CallbackResponseTransformer.java (hunk 2 above)
Comment (MarcusHMCTS) [reply src/main/java/uk/gov/hmcts/probate/transformer/CallbackResponseTransformer.java:442]: updated

File: src/main/java/uk/gov/hmcts/probate/service/CaseEscalatedService.java (hunk 3 above)
Comment (MarcusHMCTS) [reply src/main/java/uk/gov/hmcts/probate/service/CaseEscalatedService.java]: i find the is the pattern here, the service usually update the case_data
and the state is updated by the transformer

**Review by DelythJustice — APPROVED** [review]

**Pull request closed without merging**